ndcube/asdf/resources/schemas/NDCube-0.1.0.yaml

```diff
@@ -0,0 +1,25 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "asdf://sunpy.org/ndcube/schemas/NDCube-0.1.0"
+
+title:
+  Represents the ndcube NDCube object
+
+description:
+  Represents the ndcube NDCube object
+
+type: object
+properties:
+  data:
+    tag: "tag:stsci.edu:asdf/core/ndarray-1.*"
+  wcs:
+    tag: "tag:stsci.edu:gwcs/wcs-1.*"
+  extra_coords:
+    $ref: "asdf://sunpy.org/ndcube/schemas/ExtraCoords-0.1.0"
+  globa_coords:
+    $ref: "asdf://sunpy.org/ndcube/schemas/GlobalCoords-0.1.0"
+
+required: [data , wcs]
+allowAdditionalProperties: False
```

Comment:
If I subclass NDCube and want to write my own schema for my subclass with additional properties does this get in the way? @braingram

Reply:
I think so. A typical way to "subclass" a schema is to `$ref` the parent schema in a `allOf`. Using `allowAdditionalProperties: False` here would prevent the sub-schema from adding properties (because any addition would cause the parent schema validation to fail, causing the `allOf` to fail).